bin/main.ml

```diff
@@ -139,6 +142,35 @@ module Scheduler = struct
       fiber
     in
     Scheduler.go ?log ~config:common.config fiber
+
+  let poll ?log ~common ~once ~wait_success ~wait_failure =
+    let once =
+      Main.set_concurrency ?log common.config
+      >>= fun () ->
+      once ()
+    in
+    let rec main_loop () =
+      once
+      >>= fun _ ->
+      wait_success ()
+      >>= fun _ ->
+      main_loop ()
+    in
+    let continue_on_error () =
+      wait_failure ()
+      >>= fun _ ->
+      main_loop ()
+    in
+    try
```

Comment:
We can simplify this code with a single loop: `let rec loop f = try Scheduler.go ... (f ()) ...`